radicle/src/git.rs

```diff
@@ -48,31 +55,36 @@ pub mod refs {
     pub mod storage {
         use super::*;
 
-        pub fn branch(remote: &RemoteId, branch: &str) -> String {
-            format!("refs/remotes/{remote}/heads/{branch}")
+        pub fn branch(remote: &RemoteId, branch: &RefStr) -> RefString {
+            refname!("refs/remotes")
+                .and::<RefString>(remote.into())
+                .and(refname!("heads"))
+                .and(branch)
```

Comment:
```suggestion
        pub fn branch<R>(remote: &RemoteId, branch: R) -> Qualified
        where 
            R: AsRef<RefStr> 
        {
            lit::refs_remotes(
                remote,
                component!("heads").join(branch)
            ).into()
```

The `lit` module has some common patterns to save you typing out `refs/remotes`. The `component!` macro is used for single parts of the ref path. The `AsRef<RefStr>` makes the function more malleable allowing it to accept lots of different ref paths. The `Qualified` return type indicates that this is a ref path that starts with `refs` and has a category (in this case `remotes`).

Reply:
hmm the current code reads a lot better though.

I think the `refs_remote` api could be improved to maybe:
```rust
lit::refs_remotes(remote).and(component!("heads")).and(branch)
```

Reply:
You could do that just need to convert the `refs_remote` into a `Qualified` so:
```rust
lit::refs_remotes(remote).into().and(component!("heads")).and(branch)
```